pandas/tests/io/formats/test_format.py

```diff
@@ -1417,6 +1417,18 @@ def test_to_string_float_index(self):
                     '5.0  4')
         assert result == expected
 
+    def test_to_string_complex_float_formatting(self):
+        fmt.set_option('display.precision', 5)
+        df = DataFrame({'x': [
+            (0.4467846931321966 + 0.0715185102060818j),
+            (0.2739442392974528 + 0.23515228785438969j),
+            (0.26974928742135185 + 0.3250604054898979j)]})
+        result = df.to_string()
+        expected = ('                  x\n0  0.44678+0.07152j\n'
+                    '1  0.27394+0.23515j\n'
+                    '2  0.26975+0.32506j')
+        assert result == expected
+
```

Comment:
What did this test used to output (before your changes)

Reply:
Before my changes the display.precision that is set to 5 would have been ineffective. The result would be exactly the same as the input.